text/1444-union.md

```diff
@@ -29,108 +81,72 @@ space-efficient or cache-efficient structures relying on value representation,
 such as machine-word-sized unions using the least-significant bits of aligned
 pointers to distinguish cases.
 
-The syntax proposed here recognizes `union` as though it were a keyword when
-used to introduce a union declaration, *without* breaking any existing code
-that uses `union` as an identifier.  Experiments by Niko Matsakis demonstrate
-that recognizing `union` in this manner works unambiguously with zero conflicts
-in the Rust grammar.
-
-To preserve memory safety, accesses to union fields may only occur in unsafe
-code.  Commonly, code using unions will provide safe wrappers around unsafe
-union field accesses.
-
-# Detailed design
-[design]: #detailed-design
+Drop rules for unions provide enhanced control over destroying values, so unions
+can be used for implementing types like `ManuallyDrop` which are useful for
+Rust itself and not related to FFI.
 
-## Declaring a union type
+# Overview
 
-A union declaration uses the same field declaration syntax as a struct
-declaration, except with `union` in place of `struct`.
+A union declaration uses the same syntax as a struct declaration, except with
+`union` in place of `struct`.
 
 ```rust
+#[repr(C)]
 union MyUnion {
     f1: u32,
     f2: f32,
 }
 ```
 
-By default, a union uses an unspecified binary layout.  A union declared with
-the `#[repr(C)]` attribute will have the same layout as an equivalent C union.
-
-A union must have at least one field; an empty union declaration produces a
-syntax error.
-
-## Contextual keyword
-
-Rust normally prevents the use of a keyword as an identifier; for instance, a
-declaration `fn struct() {}` will produce an error "expected identifier, found
-keyword `struct`".  However, to avoid breaking existing declarations that use
-`union` as an identifier, Rust will only recognize `union` as a keyword when
-used to introduce a union declaration.  A declaration `fn union() {}` will not
-produce such an error.
+The key property of unions is that all fields of a union share common storage.
+As a result writes to one field of a union can overwrite its other fields,
```

Comment:
If we write into an empty field, e.g. `()`, it will not overwrite anything (https://github.com/rust-lang/rfcs/pull/1897#discussion_r100723787).